### PR TITLE
Expose AccountAddress for AccountState through EventKey

### PR DESCRIPTION
**Pull request by thefallentree**
Per previous discussion, as implemented in this PR, we will expose AccountAddress through EventKey for AccountState .

### CODE REVIEW

File: types/src/event.rs [hunk 1]
```diff
@@ -64,6 +65,15 @@ impl From<&EventKey> for [u8; EventKey::LENGTH] {
...
 
+impl From<&EventKey> for AccountAddress {
+    fn from(e: &EventKey) -> Self {
+        let mut arr_bytes = [0u8; AccountAddress::LENGTH];
+        arr_bytes.copy_from_slice(&e.0[EventKey::LENGTH - AccountAddress::LENGTH..]);
+
+        AccountAddress::new(arr_bytes)
+    }
+}
+
```
Comment (msmouse) [inline types/src/event.rs]: it's not really a conversion, without too much convenience gain by implementing `From`, I'd rather have a `EventKet::get_account_address()`

Comment (thefallentree) [reply types/src/event.rs]: done!

File: types/src/event.rs [hunk 2]
```diff
@@ -10,7 +10,8 @@ use rand::{rngs::OsRng, RngCore};
 use serde::{de, ser, Deserialize, Serialize};
 use std::{convert::TryFrom, fmt};
 
-/// A struct that represents a globally unique id for an Event stream that a user can listen to.e
+/// A struct that represents a globally unique id for an Event stream that a user can listen to.
+/// By design, the lower part of EventKey is the same as account address.
```
Comment (sblackshear) [inline types/src/event.rs]: Maybe "lower" -> "last 16 bytes" for clarity?

**Comment by runtian-zhou** [comment]
I'm a bit concerned about exposing this api publicly tho. The address here technically means the "creator" of this `EventHandle`, and it might have nothing to do with the sender nor the owner of this event. Leaving this api might cause some confusions there.

**Comment by sblackshear** [comment]
Would renaming `EventKey::get_account_address` to `EventKey::get_creator_address` address your concern? 

I agree with your concerns re: `EventKey` (although without open publishing we can ensure that creator address always == published address). But the `AccountState::get_address` API seems fine to me because we control the `LibraAccount` type.

**Comment by thefallentree** [comment]
> I'm a bit concerned about exposing this api publicly tho. The address here technically means the "creator" of this `EventHandle`, and it might have nothing to do with the sender nor the owner of this event. Leaving this api might cause some confusions there.

Are you saying we should do it directly in AccountState() , not in  the EventKey ?  since the purpose of this PR is to get AccountAddress for the AccountState.

**Comment by thefallentree** [comment]
updated the name as get_creator_address()

**Pull request closed without merging**